### ISSUE TITLE
Add methods push, pop, shift, unshift to array

### ISSUE THREAD

**Issue by u16suzu** [robb/Underscore.m#31]
Hi I'm u16suzu, from CA, US.

Thanks for good library.
I'm using your library for my private project.

BTW, I want to add these methods on my issues's title.
Actually theses are Ruby's methods, not but Underscore.js.

push: add element to tail of array
pop: remove element from tail of array

unshift: add element to head of array
shift: remove element from head of array

Thanks for read.

**Comment by robb** [comment]
Well, the thing about `pop` and `shift` in Ruby is that they modify the array you call the methods on. This is conceptually different from Underscore.m which leaves the wrapped array unaffected (it may not even be mutable…).

Consider

``` ruby
array = [1, 2, 3]
array.pop // => 3
array     // => [1, 2]
```

What is it, that you're after?
- If you are after the the elements returned by `pop` and `shift` – `3` in the example – you can already achieve that effect with `head` and `tail` (or `first` and `last` if you want single elements).
- If you'd like the other elements – `[1, 2]` in the example – you'd use `drop` as described in #29 for `shift` and something like underscore.js' `initial` method which is something like a `dropRight`…

As for `push` and `unshift`, I feel that `prepend` and `append` that take NSArrays would be better solutions. What do you think?

**Comment by u16suzu** [comment]
Hi, robb

Yes, theses method in Ruby I wrote are all destructive method. Theses modify itself.
Actually, I wanna  both of your suggestions(That's ruby style).

But, I understood that conceptually different between Underscore.m and Ruby.
I have used to Ruby's style so, I felt a sense of incongruity. 

OK. I will use both methods head and drop.
I have not  knew that I can use these methods. It's useful.

Thank you teach for me!
